### PR TITLE
storage: Rename "Name" field to "Filesystem label" and move it down

### PR DESCRIPTION
**Pull request by mvollmer**
Both partitions and filesystems can have "names", and people don't know that this field is for when creating a partition.  Also, giving a filesystem a label is optional and not very important, so having it at the top makes it seem more important than it is.

https://bugzilla.redhat.com/show_bug.cgi?id=2264540 
https://github.com/cockpit-project/cockpit/issues/19170

### CODE REVIEW

**Comment by garrett** [comment]
Ah, nice!

I was going to suggest moving it down and changing it to "Display name" as I thought it isn't always a label? You beat me to it! Thanks!

I'm not entirely happy with "Filesystem label" (due to jargon), but if that is what it is and what people would expect, then I guess it works.

GNOME Disks calls it "Volume Name", but I don't think that's necessarily an improvement over "Name". 

image

KDE just uses "Label":

image
image

Windows calls it "Volume label":

image

macOS calls it "Partition" or "Name" (it seems "Partition" is the old name and "Name" is the new name):


image
image
image
image

What's interesting to note: 
1. GNOME doesn't ask for a mount point. But it's a desktop and when you mount it, it uses the label as the name in /run/media/$USER/$LABEL
2. GNOME uses a 2-page (or 3 page when you select "other") wizard
3. Windows uses a Wizard
4. macOS doesn't ask for a mount point (but it's a desktop OS too, just like GNOME)
5.  Windows _does_ ask for a mount point in their wizard, but their "mount points" are alphabetic drive letters.
6. While macOS calls it "Name" in newer versions, it's directly under a "Partition Information" heading, so it's implied to be a "partition name".

**Comment by mvollmer** [comment]
> I was going to suggest moving it down and changing it to "Display name" as I thought it isn't always a label?

Oh, I only now realize that the dialog doesn't say "Filesystem" anywhere else. What about saying "filesystem" also in the "Type" dropdown and having the label immediately below "Type":

image

Ultimately, I think we should have a dedicated dialog just for formatting filesystems. Other things like erasing a block device, formatting it as swap, or creating a biosboot partition should have their own dialogs.

**Comment by garrett** [comment]
First pass at redesigning the dialog:

image

Changes:

1. Reduce default complexity
2. Reorder based on importance and commonality (all partitions have a type; most have an interactive size; the type possibly influences everything below, including size in the case of biosboot)
3. Group encryption and disk wiping in an optional area (expander in this mockup)
4. When clicking on security, it should scroll the security area to be fully in-screen (which includes the wiping/zeroing/secure-erase option)
5. Change the encryption dropdown from a dropdown to something else (as it has only 3 items); here it's a radio group
6. Make the number input formatted like a number value (monospace, not proportional)

I guess even if we keep the secure erase (it's not really useful; if people really care about security, they'd use encrypted partitions already), then the expander might still be overkill, as the encryption options are already hidden unless selected. Additionally, I'm wondering if we want mount point or the label first. I guess it depends on which one is more likely to be used? As people can mount things in arbitrary locations, we could suggest a label based on the last part of the path (/foo/bar => Bar, /my/wacky/path/ => Path). I'd suggest capitalizing, but that wouldn't work for every language and might get tricky, so perhaps that'd be "bar" and "path" as the names still, unless we have a function to properly capitalize that works across languages.

image

Anyway, I'm trying to figure out how much we would want to do to address the labeling and the amount of scrolling in one go.

Basically this is trying to cover this PR and https://github.com/cockpit-project/cockpit/pull/20060 and a bit more, taking a step back at the dialog in general.

What do you think? It's all up for discussion. I just wanted to present a few ideas together like this and get your feedback.

**Comment by garrett** [comment]
Note, it's even simpler if we drop "secure erase", as:

1. It wears down SSDs/NVMEs
2. It isn't really effective if it's done via software
3. Trim itself triggers built-in secure erase
4. People should be using encryption by default for confidential files, not relying on formatting it later
5. At least in Anaconda, people are intending to re-use the partition anyway, where data will be written to it
6. Hardware level secure erase (which happens with trim) is not only better, but also faster than writing 0s ... as writing 0s doesn't even really work on SSDs/NVMEs: https://www.makeuseof.com/tag/securely-erase-ssd-without-destroying/

I'm saying it's a relic from the past which doesn't really apply in modern times.

If we want to continue supporting erasing a partition, perhaps we should move it out from create and into delete or a special "wipe" option which is different considering what the storage device supports.

**Comment by mvollmer** [comment]
Continued in #19882
